lib/jaeger/client/udp_sender.rb

```diff
@@ -67,6 +68,14 @@ def emit_batch(thrift_spans)
         )
 
         @client.emitBatch(batch)
+      rescue StandardError => error
+        log_error(error, batch)
+      end
+
+      def log_error(error, failed_batch)
+        return unless @logger
```

Comment:
I don't think we need this line. If somebody wants to disable the logger then they can just use `Logger.new('/dev/null')` or they can raise the logging level. I'd like to avoid unnecessary nil checks. Without this I think it's reasonable to inline the log statement on line 72.

Reply:
Sure :)